android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/permission/GeckoSitePermissionsStorage.kt

```diff
@@ -340,13 +341,16 @@ class GeckoSitePermissionsStorage(
     @VisibleForTesting
     internal suspend fun clearGeckoCacheFor(origin: String) {
         withContext(mainScope.coroutineContext) {
-            geckoStorage.clearDataFromHost(origin, ClearFlags.PERMISSIONS).await()
+            geckoStorage.clearDataFromHost(
+                origin,
+                Engine.BrowsingData.select(PERMISSIONS).types.toLong(),
+            ).await()
         }
     }
 
     @VisibleForTesting
     internal fun clearAllPermissionsGeckoCache() {
-        geckoStorage.clearData(ClearFlags.PERMISSIONS)
+        geckoStorage.clearData(Engine.BrowsingData.select(PERMISSIONS).types.toLong())
```

Comment:
@jackyzy823 I think we could call directly `Engine.BrowsingData.PERMISSIONS`, as[ Engine.BrowsingData.PERMISSIONS](https://github.com/mozilla-mobile/firefox-android/blob/3b2bd69f8ebfdc4b74af71162636a75967ed6556/android-components/components/concept/engine/src/main/java/mozilla/components/concept/engine/Engine.kt#L41) and [StorageController.PERMISSIONS](https://searchfox.org/mozilla-central/source/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/StorageController.java#51) are the same value (`64`). What do you think?

Reply:
Yes, but i think we could keep the same way passing flags to clearData* methods.

Reply:
Could you clarify why do we need the extra step of using `Engine.BrowsingData.select`?

Reply:
Could be use  `geckoStorage.clearData(Engine.BrowsingData.PERMISSIONS.toLong())` instead?